### PR TITLE
Feature/svcs 337 sdk supports OIDC

### PR DESCRIPTION
**Pull request by chad-pelly**

### CODE REVIEW

**Comment by chad-pelly** [comment]
I assume the user used for testing is missing the permission.

File: src/models/oidc/Providers.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+export interface BaseOidcProvider {
```
Comment (chad-pelly) [inline src/models/oidc/Providers.ts:1]: These are extractions from other services. I assume we have a reference repository ticket somewhere to some day set up a joint models package?

File: src/services/auth/oidc.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import { HttpInstance, OidcService } from '../../types';
+import { HttpClient } from '../http-client';
+
+export default (
+  oidcClient: HttpClient,
+  httpWithAuth: HttpInstance
+): OidcService => ({
+  async createProvider(requestBody) {
+    const { data } = await oidcClient.post(
```
Comment (chad-pelly) [inline src/services/auth/oidc.ts:9]: The other endpoints don't extract `{ data }` they just return x.data. This is more of a preference for me and I can quickly change it back if we don't like it. Simple code yamsayin'.

File: src/models/oidc/Providers.ts (hunk 1 above)
Comment (chad-pelly) [inline src/models/oidc/Providers.ts:1]: I also feel that the `types` file referencing all service interfaces and all models is getting cumbersome. Here I have started the implementation as if were a traditional service.

Alternative I can put it all in the `types` file and create a reference repository ticket to change that at a later date.

File: tests/unit/auth/oidc.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+  beforeAll(async () => {
+    await sdk.auth.authenticate({
+      email: process.env.API_USERNAME,
+      password: process.env.API_PASSWORD,
+    });
+  });
+
+  beforeEach(async () => {
+    const data = createProviderData();
```
Comment (chad-pelly) [inline tests/unit/auth/oidc.ts:22]: This technically would be nicer if it was declared individually in each test, especially the create. But it seemed unnecessary.

File: src/services/auth/types.ts [hunk 4]
```diff
@@ -434,3 +439,61 @@ export interface AuthOauth1Service {
...
+
+  /**
+   * Update an OpenId Connect Provider
+   * Permission | Scope | Effect
+   * - | - | -
+   * `UPDATE_OIDC_PROVIDER` | `global` | **Required** for this endpoint
+   */
+  updateProvider(
+    providerId: string,
+    requestBody: Partial<OidcProviderUpdate>
```
Comment (AaronTrippaers) [inline src/services/auth/types.ts:473]: (minor) The OidcProviderUpdate is already a partial, so this is technically not need 😄

File: tests/unit/auth/oidc.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    provider = await sdk.auth.oidc.createProvider(data);
+  });
+
+  afterAll(async () => {
+    await sdk.auth.oidc.deleteProvider(provider.id);
+  });
+
+  it('Creates an OpenID Connect provider', async () => {
```
Comment (AaronTrippaers) [inline tests/unit/auth/oidc.ts:30]: I would still do this create a call in the test here, now the test and the beforeEach are mangled.

File: src/models/oidc/Providers.ts (hunk 1 above)
Comment (wwaaijer-exh) [reply src/models/oidc/Providers.ts:1]: Yes! The idea was to generate the types based on the swagger files some day. Same goes for the swagger files themselves, generating them based on the types/validators

File: src/models/oidc/Providers.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  name: string; // Human friendly name of the provider, which can also be used in the oidc login urls
```
Comment (wwaaijer-exh) [inline src/models/oidc/Providers.ts:2]: (minor) If you move these comments to JSDoc they'll nicely show up in the editor for user of the SDK. Like:
```
/** Human friendly name of the provider, which can also be used in the oidc login urls */
name: string;
```

File: src/services/auth/oidc.ts (hunk 2 above)
Comment (wwaaijer-exh) [reply src/services/auth/oidc.ts:9]: Yeah, this is way better 👍 I've a note for this as well. Idea right now was to refactor this away to be included in the `HttpClient` logic, as this is happening in basically every file 😬

File: tests/__helpers__/oidc.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import { OidcProviderCreation } from '../../src/models/oidc/Providers';
+
+export const generateRandomName = () =>
+  `${crypto.randomBytes(5).toString('hex')}-${crypto
+    .randomBytes(5)
+    .toString('hex')}`;
+export const createProviderData = (
+  data?: Partial<OidcProviderCreation>
+): OidcProviderCreation => ({
+  name: `${crypto.randomBytes(5).toString('hex')}-${crypto
```
Comment (wwaaijer-exh) [inline tests/__helpers__/oidc.ts:11]: I assume you can reuse `generateRandomName` here? 😄

File: tests/unit/auth/oidc.ts [hunk 8]
```diff
@@ -0,0 +1,91 @@
+import { createClient, rqlBuilder } from '../../../src';
+import { createProviderData, generateRandomName } from '../../__helpers__/oidc';
+import { OidcProviderResponse } from '../../../src/models/oidc/Providers';
+
+describe('Enables an OpenID Connect workflow', () => {
```
Comment (wwaaijer-exh) [inline tests/unit/auth/oidc.ts:5]: These tests seem to be written as end 2 end tests, not unit tests

**Review by wwaaijer-exh — APPROVED** [review]
Nice nice 👍

**Pull request merged**